pkg/azure/utils.go

```diff
@@ -423,9 +429,8 @@ func (d *MachinePlugin) createVMNicDisk(req *driver.CreateMachineRequest) (*comp
 	*/
 	startTime := time.Now()
 	imageRefClass := providerSpec.Properties.StorageProfile.ImageReference
-	// if ID is not set the image is referenced using a URN
-	if imageRefClass.ID == "" {
-
+	// if ID and community id are not set the image is referenced using a URN
+	if imageRefClass.URN != nil {
```

Comment:
two questions
* following the comment , shouldn't this condition be 
`if (imageRefClass.ID is not set) && (imageRefClass.CommunityGalleryImageID is not set)` then {...}

* also I don't see the else case when community ID is set

Reply:
The actual image will be set later in `getVMParameters()`, where `getImageReference()` is called.

Here we just check if an urn is used and then fetch the image to check if it has a plan included where we need to accept the marketplace agreement (see lines below).